File: packages/audio/android/src/main/java/dev/tcode/thinmpr/audio/extension/AlbumModelExtension.kt
package dev.tcode.thinmpr.audio.extension

import dev.tcode.thinmpr.audio.model.contract.AlbumModelContract

fun AlbumModelContract.toMap(): Map<String, Any> {
  return mapOf(
    "id" to this.id.raw,
    "name" to this.name,
    "artistId" to this.artistId.raw,
    "artistName" to this.artistName,
    "imageId" to this.imageId
  )
}

File: packages/audio/ios/Extension/AlbumModelExtension.swift
extension AlbumModelContract {
    func toMap() -> [String: Any] {
        return [
            "id": String(self.id.raw),
            "name": self.name,
            "artistId": String(self.artistId.raw),
            "artistName": self.artistName,
            "imageId": self.imageId,
        ]
    }
}
